include/dsn/dist/replication/replication.types.h

```diff
@@ -469,6 +469,14 @@ namespace dsn { namespace replication {
     {
         int32_t app_id;
         int32_t pidx;
+        inline bool operator < (const global_partition_id& r) const
```

Comment:
Before I also considered the overwritten problem, but if the comparators are moved back to replication_common.h, the comparators will not be seen by users who include the "replication.types.h", and when compare global_partition_id, it would use default "byte-by-byte" comparator.